### ISSUE TITLE
equal/compare for cenums?

### ISSUE THREAD

**Issue by yomimono** [mirage/ocaml-cstruct#128]
It would be nice to automatically get `equal` or `compare` functions for types created from `cenum`s.

**Comment by avsm** [comment]
is this for satisfying a particular function signature? (i.e. do you need just `compare` or is `equal` also definitely needed).  Since they are just integers, polymorphic comparison is normally fine, but generating explicit comparators is definitely a good thing.

**Comment by yomimono** [comment]
In short, stuff potentially from cenums -> record -> alcotest .

**Comment by avsm** [comment]
This sounds like having `compare` is enough -- I'm a little wary of generating too many functions from the ppx.

**Comment by yomimono** [comment]
Sure, `compare` is probably fine.

**Comment by emillon** [comment]
N.B: supposing that you want to compare by id, polymorphic comparison gives the wrong results for cases like this:

```ocaml
[%%cenum
type reversed = 
  | ONE_R [@id 2]
  | TWO_R [@id 1]
  [@@uint8_t]
] 
```

I'll have the ppx handle this, but that could be a source of bugs.

**Comment by avsm** [comment]
good catch!

**Comment by avsm** [comment]
fixed, will be in cstruct 5.0+
